video/out/vo_vaapi.c

```diff
@@ -684,6 +684,9 @@ static void draw_osd(struct vo *vo)
         int rw = mp_rect_w(*rc);
         int rh = mp_rect_h(*rc);
 
+        if (n == num_mod_rc - 1)
```

Comment:
Could you just add a brief comment explaining this is a special case? Also in the commit message if you could change `Fixes issue 10436` to `Fixes issue #10436` so github autolinks it, that would be great.